### DISCUSSION TITLE
[Feature Request] Getting the error message in a pop out instead of displaying inside the editor

### DISCUSSION THREAD

**Discussion by Ananya-Gupta24** [microsoft/monaco-editor#3691] [Q&A]
### Context

- [X] This issue is not a bug report. *(please use a different template for reporting a bug)*
- [X] This issue is not a duplicate of an existing issue. *(please use the [search](https://github.com/microsoft/monaco-editor/issues) to find existing issues)*

### Description

When I run this Hello World! code sample (https://microsoft.github.io/monaco-editor/playground.html#creating-the-editor-hello-world)

```
monaco.editor.create(document.getElementById('container'), {
    value: "function hello() {\n\talert('Hello world!');\n}",
    language: 'javascript'
});
```
and change "function" to say "functionn", I get a suggestion box with a "View the Problem" link. When I click it, it renders the error inside a box within the editor like so-

![popout](https://user-images.githubusercontent.com/92943978/155179214-79786b6d-2c60-47d7-8d90-9cf677ccda69.PNG)

Is it possible to render the error in a popout, instead of inside the editor (for scenarios where the editor size is small)?

**Comment by hediet** [comment]
Hey Ananya! There is no pop-up mechanism that I know of that could help you here out of the box.

You could directly use decorations instead of diagnostics. Then you are in control of rendering the squiggles. This could be used as a base for your own pop-up when the user hovers/clicks on these error decorations.


I think this approach is easier than using diagnostics + changing the way errors are presented in the editor widget.

Related: https://github.com/microsoft/monaco-editor/issues/2537
